src/app/Shared/Services/Api.service.tsx

```diff
@@ -289,6 +291,28 @@ export class ApiService {
     );
   }
 
+  createSnapshotV2(): Observable<Recording> {
+    return this.target.target().pipe(
+      concatMap((target) =>
+        this.sendRequest('v2', `targets/${encodeURIComponent(target.connectUrl)}/snapshot`, {
+          method: 'POST',
+        }).pipe(
+          concatMap((resp) => {
+            if (resp.status == 202) {
+              return throwError(() => new Error('Unable to create snapshot'));
+            } else if (resp.status == 427) {
```

Comment:
These errors (i.e. 427) are already handled within`sendRequest`. Additional handling (i.e. 202 code), should be done in outer catchError here (i.e. use isHttpError) and return perhaps `of(undefined)` (need to change function signature).

Also, I think `createSnapshotV2()` is called by an action button. Generally, this should not throw any error (might be caught and show error view) because usually we want the UI to stay to retry.

Reply:
createSnapshotV2 actually isn't being called by anything right now. I replaced using snapshots, but kept the function in case of future use. I will remove the error throwing then.